service/popm/popm.go

```diff
@@ -742,7 +742,8 @@ func (m *Miner) handleBFGCallCompletion(parrentCtx context.Context, conn *protoc
 	select {
 	case bc.ch <- payload:
 		log.Tracef("handleBFGCallCompletion returned: %v", spew.Sdump(payload))
-	default:
+	case <-time.After(m.requestTimeout):
```

Comment:
hey @AL-CT one more thing we might want to add here is a `case` for the context being "done" 

perhaps something like (double check this)
```go
case <-ctx.Done():
```

not a big deal, since we will timeout anyway, but we probably should have this